keps/sig-storage/361-local-ephemeral-storage-isolation/README.md

```diff
@@ -0,0 +1,471 @@
+
+
+##### Unit tests
+
+<!--
+In principle every added code should have complete unit test coverage, so providing
+the exact set of tests will not bring additional value.
+However, if complete unit test coverage is not possible, explain the reason of it
+together with explanation why this is acceptable.
+-->
+
+<!--
+Additionally, for Alpha try to enumerate the core package you will be touching
+to implement this enhancement and provide the current unit coverage for those
+in the form of:
+- <package>: <date> - <current test coverage>
+The data can be easily read from:
+https://testgrid.k8s.io/sig-testing-canaries#ci-kubernetes-coverage-unit
+
+This can inform certain test coverage improvements that we want to do before
+extending the production code to implement this enhancement.
+-->
+
+- `<package>`: `<date>` - `<test coverage>`
+eviction: pkg/kubelet/eviction/helpers_test.go#L702
+eviction: pkg/kubelet/eviction/eviction_manager_test.go#L452
+cm: pkg/kubelet/cm/container_manager_linux_test.go#L243
+scheduler: pkg/scheduler/framework/plugins/noderesources/fit_test.go
+
+##### Integration tests
+
+<!--
+This question should be filled when targeting a release.
+For Alpha, describe what tests will be added to ensure proper quality of the enhancement.
+
+For Beta and GA, add links to added tests together with links to k8s-triage for those tests:
+https://storage.googleapis.com/k8s-triage/index.html
+-->
+
+- <test>: <link to test coverage>
+
+##### e2e tests
+
+<!--
+This question should be filled when targeting a release.
+For Alpha, describe what tests will be added to ensure proper quality of the enhancement.
+
+For Beta and GA, add links to added tests together with links to k8s-triage for those tests:
+https://storage.googleapis.com/k8s-triage/index.html
+
+We expect no non-infra related flakes in the last month as a GA graduation criteria.
+-->
+
+- <test>: <link to test coverage>
+LocalStorageCapacityIsolationEviction: https://github.com/kubernetes/kubernetes/blob/master/test/e2e_node/eviction_test.go#L289
+ephemeral storage resource limits: https://github.com/kubernetes/kubernetes/blob/master/test/e2e/scheduling/predicates.go
+
+
+## Graduation Criteria
+
+The following criteria applies to
+`LocalStorageCapacityIsolation`:
+
+### Phase 1: Alpha (1.8)
+
+- basic support added in node/core APIs 
+- support integrated in kubelet
+- Alpha-level documentation
+- Unit test coverage
+
+### Phase 2: Beta
+
+- node e2e test coverage
+- monitoring matric
+
+### Phase 3: GA (1.25)
+
+- user feedback based on production experience
+
+### Upgrade / Downgrade Strategy
+
+Feature is enabled by default since 1.10
+If feature is disabled when creating or updating cluster, in case of
+request is set for ephemeral storage, scheduler will not take this ephemeral
+storage request into consideration when scheduling pod. If limit is set, pod
+will not be evicted due to ephemeral storage usage exceeding limit. sizeLimit
+for emptyDir will also not be enforced.
```

Comment:
So, if someone had set a limit, but the feature was disabled, the limit would have been ignored in the past. If they upgrade, the feature is now locked on. So, will any existing pods that have limits exceeded be evicted?

If so, that's not a blocker - but it IS something that needs to go in the release notes and highlighted as an upgrade risk.

The answer to "will this affect running workloads" should be YES, in the event that the feature was previously disabled. Imagine upgrading and all of a sudden a bunch of Pods get evicted and rescheduled (hopefully!)...